python/pyspark/ml/util.py

```diff
@@ -536,10 +536,8 @@ def __get_class(clazz: str) -> Type[RL]:
         """
         parts = clazz.split(".")
         module = ".".join(parts[:-1])
-        m = __import__(module)
-        for comp in parts[1:]:
-            m = getattr(m, comp)
-        return m
+        m = __import__(module, fromlist=[parts[-1])
```

Comment:
I think it depends on which mypy version we use. cc @Yikun FYI.
If we can find a way to make all versions of mypy happy, that would be ideal.

Reply:
I also take a deep look on this:

> When the name variable is of the form package.module, normally, the top-level package (the name up till the first dot) is returned, not the module named by name. However, when a non-empty fromlist argument is given, the module named by name is returned.

such for `pyspark.ml.classification.LinearSVC`

```python
>>> clazz = "pyspark.ml.classification.LinearSVC";parts = clazz.split(".");module = ".".join(parts[:-1])
>>> __import__(module)
<module 'pyspark' from '/Users/yikun/spark/python/pyspark/__init__.py'>
>>> __import__(module, fromlist=[parts[-1]])
<module 'pyspark.ml.classification' from '/Users/yikun/spark/python/pyspark/ml/classification.py'>
```

and also return same results of wrong name like: `pyspark.ml.classification.unknow`, `pyspark.ml.unknow.LinearSVC`

After this, import with non-empty fromlist works well, can also simplified implementations and cleanup the mypy error:
```
python/pyspark/ml/util.py:542: error: Incompatible return value type (got Module, expected "Type[RL]")  [return-value]
```

So, I'm fine with this change.